hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/CreatePipelineCommandHandler.java

```diff
@@ -67,6 +75,10 @@ public CreatePipelineCommandHandler(ConfigurationSource conf) {
   CreatePipelineCommandHandler(
       BiFunction<RaftPeer, GrpcTlsConfig, RaftClient> newRaftClient) {
     this.newRaftClient = newRaftClient;
+    this.executor = Executors.newFixedThreadPool(
+        1, new ThreadFactoryBuilder()
+            .setNameFormat("CreatePipelineCommandHandlerThread-%d").build());
+    this.workQueue = ((ThreadPoolExecutor) this.executor).getQueue();
```

Comment:
Should we share the same single-threaded executor between `CreatePipelineCommandHandler` and `ClosePipelineCommandHandler`?

Also, would be nice to use `MoreExecutors.directExecutor()` for unit tests (`TestCreatePipelineCommandHandler`).  Otherwise the test needs to be updated to wait for the expected results, since event is now handled in another thread.

Reply:
@adoroszlai Thanks. That's a good idea. I'll improve it